### PR TITLE
Added image for building Istio proxy image based on cilium/envoy

### PR DESCRIPTION
**Pull request by mmnelemane**
Proxy image required by https://github.com/SUSE/avant-garde/issues/1510

### CODE REVIEW

File: caasp-istio-pilot-image/_service [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <services>
     <service name="replace_using_package_version" mode="buildtime">
         <param name="file">caasp-istio-pilot-image.kiwi</param>
-        <param name="regex">%%PKG_VERSION%%</param>
+        <param name="regex">1.5.4</param>
```
Comment (davidcassany) [inline caasp-istio-pilot-image/_service:4]: This makes no sense. If for some reason the tag you need does not match the version of any of the installed packages installed within the image, then this service is not needed. If istio is in version `1.5.4` then please use the service and you'll get that automatically aligned on following istio updates without having to submit image source changes just to bump de version.

File: caasp-istio-pilot-image/caasp-istio-pilot-image.kiwi [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 
-<!-- OBS-AddTag: caasp/v4/istio-pilot:%%PKG_VERSION%%-rev<VERSION> caasp/v4/istio-pilot:%%PKG_VERSION%%-rev<VERSION>-build<RELEASE> -->
+<!-- OBS-AddTag: caasp/v4/istio-pilot:1.5.4-->
```
Comment (davidcassany) [inline caasp-istio-pilot-image/caasp-istio-pilot-image.kiwi:3]: Why this change? Please use the same tagging schema of other images. Note that %%PKG_VERSION%% placeholder will be replaced by istio version if you have a _service like:
```
<service name="replace_using_package_using">
    <param name="file">caasp-istio-pilot-image.kiwi</param>
    <param name="regex">%%PKG_VERSION%%</param>
    <param name="parse-version">patch</param>
    <param name="package">istio</param>
</service>

File: caasp-istio-pilot-image/caasp-istio-pilot-image.kiwi [hunk 3]
```diff
@@ -14,7 +14,7 @@
       derived_from="obsrepositories:/suse/sle15#15.1">
       <containerconfig
         name="caasp/v4/istio-pilot"
-        tag="%%PKG_VERSION%%"
+        tag="1.5.4"
```
Comment (davidcassany) [inline caasp-istio-pilot-image/caasp-istio-pilot-image.kiwi:17]: If you use hardcoded tags for whatever reason there is no need to define the
`<service name="replace_using_package_using">` section in _service file.

File: caasp-istio-proxyv2-image/Makefile [hunk 4]
```diff
@@ -0,0 +1,10 @@
+mkfile_path := $(abspath $(lastword $(MAKEFILE_LIST)))
```
Comment (davidcassany) [inline caasp-istio-proxyv2-image/Makefile:1]: This make file should be a symlink to `../.images.Makefile` as it is for the other images.

File: caasp-istio-proxyv2-image/caasp-istio-proxyv2-image.kiwi [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  </description>
+  <preferences>
+    <type
+      image="docker"
+      derived_from="obsrepositories:/suse/sle15#15.1">
+      <containerconfig
+        name="caasp/v4/istio-proxyv2"
+        tag="1.5.4"
+        maintainer="SUSE Containers Team &lt;containers@suse.com&gt;">
+        <entrypoint execute="/usr/bin/cilium-envoy"/>
```
Comment (manuelbuil) [inline caasp-istio-proxyv2-image/caasp-istio-proxyv2-image.kiwi:19]: The entrypoint in the upstream image is different, it is:
            "Entrypoint": [
                "/usr/local/bin/pilot-agent"
            ],

File: caasp-istio-proxyv2-image/_service [hunk 6]
```diff
@@ -0,0 +1,10 @@
+<services>
+    <service name="replace_using_package_version" mode="buildtime">
```
Comment (davidcassany) [inline caasp-istio-proxyv2-image/_service:2]: The same as above, this service makes no sense if you hardcode the tags. Note the this service can query the version of any of the packages installed within the image, even if they are not listed in the description file (not sure if this is the case here).

File: caasp-istio-proxyv2-image/_service [hunk 7]
```diff
@@ -0,0 +1,10 @@
...
+        <param name="file">caasp-istio-proxyv2-image.kiwi</param>
+        <param name="regex">1.5.4</param>
```
Comment (davidcassany) [inline caasp-istio-proxyv2-image/_service:4]: Note that doing so will replace any `1.5.4` reference to the version of the `cilium-istio-proxy` package... I doubt this is intented

File: caasp-istio-proxyv2-image/caasp-istio-proxyv2-image.kiwi [hunk 8]
```diff
@@ -0,0 +1,45 @@
+<?xml version="1.0" encoding="utf-8"?>
+
+<!-- OBS-AddTag: caasp/v4/istio-proxyv2:1.5.4 -->
```
Comment (davidcassany) [inline caasp-istio-proxyv2-image/caasp-istio-proxyv2-image.kiwi:3]: Please use the three tags approach. Also note that this tag is redundant, you already defined that one in line 17.

File: caasp-istio-proxyv2-image/caasp-istio-proxyv2-image.kiwi (hunk 5 above)
Comment (mmnelemane) [reply caasp-istio-proxyv2-image/caasp-istio-proxyv2-image.kiwi:19]: @manuelbuil Thats the entrypoint for pilot image. This kiwi file is for proxyv2 image and cilium-envoy is installed in /usr/bin

Comment (manuelbuil) [reply caasp-istio-proxyv2-image/caasp-istio-proxyv2-image.kiwi:19]: Not exactly. Remember the proxyv2 container executes two processes:

```
4     0     1     0  20   0 153468 47844 -      Ssl  ?          0:01 /usr/local/bin/pilot-agent proxy router --domain istio-system.svc.cluster.local --proxyLogLevel=warning --proxyComponentLogLevel=misc:error --
4     0    15     1  20   0 169572 49796 SyS_ep Sl   ?          0:00 /usr/local/bin/envoy -c /etc/istio/proxy/envoy-rev0.json --restart-epoch 0 --drain-time-s 45 --parent-shutdown-time-s 60 --service-cluster ist
```
I can confirm you that the upstream image `docker.io/cilium/istio_proxy` has the entrypoint `/usr/local/bin/pilot-agent`

**Comment by kkaempf** [comment]
How about moving this to SLES 15 SP2 / CaaSP v5 ?

File: caasp-istio-proxyv2-image/caasp-istio-proxyv2-image.kiwi (hunk 5 above)
Comment (manuelbuil) [reply caasp-istio-proxyv2-image/caasp-istio-proxyv2-image.kiwi:19]: If I understand correctly, pilot-agent is the one triggering cilium-envoy. We need to investigate how exactly it does this

Comment (manuelbuil) [reply caasp-istio-proxyv2-image/caasp-istio-proxyv2-image.kiwi:19]: I can confirm this. pilot-agent expects the envoy binary to be `/usr/local/bin/envoy` ==> https://github.com/istio/istio/blob/master/pkg/config/constants/constants.go#L46.

I suggest doing `ln -s /usr/bin/cilium-envoy /usr/local/bin/envoy`

File: caasp-istio-proxyv2-image/Makefile (hunk 4 above)
Comment (mmnelemane) [reply caasp-istio-proxyv2-image/Makefile:1]: @davidcassany Thanks for point this out. Will fix in the next update.

File: caasp-istio-proxyv2-image/_service (hunk 6 above)
Comment (mmnelemane) [reply caasp-istio-proxyv2-image/_service:2]: the image will now be reverted to use %%PKG_VERSION%% and the cilium-istio-proxy package is now being modified to build as version 1.5.4. So this hard-coded tagging can be removed and _service file can be retained.

File: caasp-istio-proxyv2-image/caasp-istio-proxyv2-image.kiwi (hunk 5 above)
Comment (mmnelemane) [reply caasp-istio-proxyv2-image/caasp-istio-proxyv2-image.kiwi:19]: @manuelbuil Updated the entrypoint to use pilot-agent and the cilium-envoy is copied to /usr/local/bin/envoy in the package. Thanks.

File: caasp-istio-pilot-image/_service (hunk 1 above)
Comment (mmnelemane) [reply caasp-istio-pilot-image/_service:4]: the image will now be reverted to use %%PKG_VERSION%% and the cilium-istio-proxy package is now being modified to build as version 1.5.4. So this hard-coded tagging can be removed and _service file can be retained.

File: caasp-istio-pilot-image/caasp-istio-pilot-image.kiwi (hunk 2 above)
Comment (mmnelemane) [reply caasp-istio-pilot-image/caasp-istio-pilot-image.kiwi:3]: the image will now be reverted to use %%PKG_VERSION%% and the cilium-istio-proxy package is now being modified to build as version 1.5.4. So this hard-coded tagging can be removed and _service file can be retained.

File: caasp-istio-proxyv2-image/_service (hunk 7 above)
Comment (mmnelemane) [reply caasp-istio-proxyv2-image/_service:4]: the image will now be reverted to use %%PKG_VERSION%% and the cilium-istio-proxy package is now being modified to build as version 1.5.4. So this hard-coded tagging can be removed and _service file can be retained.

File: caasp-istio-proxyv2-image/caasp-istio-proxyv2-image.kiwi (hunk 8 above)
Comment (mmnelemane) [reply caasp-istio-proxyv2-image/caasp-istio-proxyv2-image.kiwi:3]: Noted. Will be updated in the next push. Thanks.

**Comment by mmnelemane** [comment]
> How about moving this to SLES 15 SP2 / CaaSP v5 ?

Moved to SLES 15 SP2 /CaaSP v5

File: caasp-istio-proxyv2-image/caasp-istio-proxyv2-image.kiwi [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+  </description>
+  <preferences>
+    <type
+      image="docker"
+      derived_from="obsrepositories:/suse/sle15#15.2">
+      <containerconfig
+        name="caasp/v5/istio-proxyv2"
+        tag="%%PKG_VERSION%%"
+        maintainer="SUSE Containers Team &lt;containers@suse.com&gt;">
+        <entrypoint execute="/usr/bin/pilot-agent proxy --binaryPath /usr/bin/envoy"/>
```
Comment (manuelbuil) [inline caasp-istio-proxyv2-image/caasp-istio-proxyv2-image.kiwi:19]: the `proxy` arg is already provided by the manifest. 
The manifest also provides a `--binaryPath`, therefore, I suggest looking for an alternative

Comment (manuelbuil) [reply caasp-istio-proxyv2-image/caasp-istio-proxyv2-image.kiwi:19]: I updated the image in IBS to have:
<entrypoint execute="/usr/bin/pilot-agent"/>

Comment (mmnelemane) [reply caasp-istio-proxyv2-image/caasp-istio-proxyv2-image.kiwi]: Updated to `/usr/bin/pilot-agent`

**Pull request merged**